### PR TITLE
Remove JSch from server module

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
Using Apache MINA instead of JSch is recommended by JGit team.

Modification:
- Use Apache MINA instead of JSch for server module.
  - JSch is now only used for testing.

Result:
- JSch gone.

### CODE REVIEW

File: dependencies.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #     If its classes are exposed in Javadoc, update offline links as well.
 #
 [versions]
-armeria = "1.25.1"
+armeria = "1.24.3"
```
Comment (minwoox) [inline dependencies.toml:6]: Will change once 1.25.1 is released.

**Comment by minwoox** [comment]
I believe the build will succeed after https://github.com/line/centraldogma/pull/867 is merged.

File: dist/build.gradle [hunk 2]
```diff
@@ -90,8 +90,8 @@ task copyLib(group: 'Build',
         FileCollection jgits = layout.files { libDir.listFiles() }.filter { File file ->
             file.name.contains("jgit")
         }
-        // org.eclipse.jgit-6.x and org.eclipse.jgit.ssh.jsch-6.x are required.
-        if (jgits.size() != 2 || !jgits.any { File file -> file.name.startsWith("org.eclipse.jgit-6") }) {
+        // org.eclipse.jgit-6.x is required.
+        if (jgits.size() != 1 || !jgits[0].name.startsWith("org.eclipse.jgit-6")) {
```
Comment (jrhee17) [inline dist/build.gradle:94]:
```suggestion
        if (jgits.size() != 1 || !jgits.first().name.startsWith("org.eclipse.jgit-6")) {
```

**Review by jrhee17 — APPROVED** [review]
Thanks @minwoox 🙇 👍 🙇

**Review by ikhoon — APPROVED** [review]
Thanks!

File: dist/build.gradle [hunk 3]
```diff
@@ -90,8 +90,8 @@ task copyLib(group: 'Build',
         FileCollection jgits = layout.files { libDir.listFiles() }.filter { File file ->
             file.name.contains("jgit")
         }
-        // org.eclipse.jgit-6.x and org.eclipse.jgit.ssh.jsch-6.x are required.
-        if (jgits.size() != 2 || !jgits.any { File file -> file.name.startsWith("org.eclipse.jgit-6") }) {
+        // org.eclipse.jgit-6.x is required.
+        if (jgits.size() != 1 || !jgits.first().name.startsWith("org.eclipse.jgit-6")) {
             throw new IllegalStateException("Unexpected JGit JARs: $jgits.files")
```
Comment (ikhoon) [inline dist/build.gradle:95]: By the way, could we move `startsWith("org.eclipse.jgit-6")` to `FileCollection`'s filter and check only if the size of `jgits` is one?

Comment (ikhoon) [reply dist/build.gradle:95]: Ah, it seems intentionally used for leaving an exception message for the jar files.
Please ignore the commend above.

**Review by trustin — APPROVED** [review]
:+1:

**Pull request merged**